### PR TITLE
New roles for broadcast, report and business process management

### PR DESCRIPTION
**Pull request by RikaDenia**
This PR adds the roles

- authorized_for_broadcasts
- authorized_for_reports
- authorized_for_business_processes

for managing broadcasts, reports and business processes respectively.

For broadcasts and reports, access is completely prohibited unless the accessing user has the correct role.
For business processes, we continue the existing behaviour and allow read access in every case, though creating / editing processes requires the authorized_for_business_processes role.

This also includes the necessary modifications to the conf plugin to allow for assigning and reviewing the new roles.

### CODE REVIEW

File: cgi.cfg [hunk 1]
```diff
@@ -134,6 +134,23 @@ authorized_contactgroup_for_all_service_commands=
...
 
+# BROADCAST ACCESS
+# A comma-delimited list of usernames that can manage broadcasts.
+authorized_for_broadcasts=thrukadmin
+authorized_contactgroup_for_broadcasts=
+
+# REPORT ACCESS
+# A comma-delimited list of usernames that have full access to the reporting
+# feature
+authorized_for_reports=thrukadmin
```
Comment (sni) [inline cgi.cfg]: I'd say we set this to `*` because thats the way it was before.

File: lib/Thruk/Authentication/User.pm [hunk 2]
```diff
@@ -140,6 +143,12 @@ sub new {
         $self->{'roles'} = [ grep({ $_ ne 'authorized_for_read_only' } @{$self->{'roles'}}) ];
     }
 
+    # Does this user have all roles?
```
Comment (sni) [inline lib/Thruk/Authentication/User.pm]: there is no need to do this twice :-)

File: lib/Thruk/Authentication/User.pm [hunk 3]
```diff
@@ -140,6 +143,12 @@ sub new {
...
+    if (@{$self->{'roles'}} == @{$possible_roles}) {
+        # Yes he/she does. Remove read only role
+        $self->{'roles'} = [ grep({ $_ ne 'authorized_for_read_only' } @{$self->{'roles'}}) ];
+    }
+
```
Comment (sni) [inline lib/Thruk/Authentication/User.pm]: i guess we have to add the new roles here to anyone who is a admin (using the logic if someone has both, authorized_for_system_information and authorized_for_configuration_information). Or we do this check later. Anyway, anyone with those two roles must still be able to access broadcasts, reports and BPs.

File: cgi.cfg (hunk 1 above)
Comment (RikaDenia) [reply cgi.cfg]: Done. See 3d69a5815cd949ce3dcaef2916aca2a1a57674c3

File: lib/Thruk/Authentication/User.pm (hunk 2 above)
Comment (RikaDenia) [reply lib/Thruk/Authentication/User.pm]: Whoops =D Removed it

File: lib/Thruk/Authentication/User.pm (hunk 3 above)
Comment (RikaDenia) [reply lib/Thruk/Authentication/User.pm]: Sounds like a good idea. Done, please see 73fcdbc9444a46aec5f49fe1daa51cd2afee8f6f

**Comment by sni** [comment]
great, looks good now. Thanks

**Pull request merged**